role/api.py

```diff
@@ -0,0 +1,50 @@
+"""
+Functions for handling roles
+"""
+
+from __future__ import unicode_literals
+
+from django.contrib.auth.models import Group
+from guardian.shortcuts import assign_perm
+
+from role.permissions import RepoPermission, GroupTypes
+
+
+def roles_init_new_repo(repo):
+    """
+    Create new groups for the repo
+    """
+    administrator_group = Group.objects.get_or_create(
```

Comment:
With certain SQL backends, *cough* MySQL *cough* we have a lot of experience (several times a day) with that constraint not working in a distributed app tier environment, so it is front of mind.